### PR TITLE
Azure Stack availability set

### PR DESCRIPTION
**Pull request by patrickdillon**
The MAO does not have the ability to create availability sets in Azure Stack Hub, so it depends on the one created by the installer. 

- renames the availability set to indicate it is used by all nodes, not just masters
- switches the AzureMachineProviderSpec to use the one from openshift/api to get new `availabilityset` field
- Remove cluster-api-provider-azure as much as possible as it is no longer needed

### CODE REVIEW

**Comment by patrickdillon** [comment]
cc @alexander-demichev

File: go.mod [hunk 1]
```diff
@@ -101,7 +101,6 @@ require (
 	k8s.io/klog/v2 v2.9.0
 	k8s.io/utils v0.0.0-20210819203725-bdf08cb9a70a
 	sigs.k8s.io/cluster-api-provider-aws v0.0.0
-	sigs.k8s.io/cluster-api-provider-azure v0.0.0
```
Comment (staebler) [inline go.mod:104]: Please separate vendor changes into their own commit.

**Review by staebler — COMMENTED** [review]
Looks good to me other than separating the vendoring changes.
/approve

/test e2e-azure

**Comment by patrickdillon** [comment]
> Looks good to me other than separating the vendoring changes. /approve

Done. Vendoring has been split up.

**Review by staebler — APPROVED** [review]
/lgtm
/approve
